Formula/ripsecrets.rb

```diff
@@ -0,0 +1,29 @@
+class Ripsecrets < Formula
+  desc "Prevent committing secret keys into your source code"
+  homepage "https://github.com/sirwart/ripsecrets"
+  url "https://github.com/sirwart/ripsecrets/archive/v0.1.5.tar.gz"
+  sha256 "1e3d36b3892d241dfd5e9abd86ddb47f22e6837b89cf9ee44989d6c1271dda2b"
+  license "MIT"
+  head "https://github.com/sirwart/ripsecrets.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    # Generate a real-looking key
+    keyspace = "A".upto("Z").to_a + "a".upto("z").to_a + "0".upto("9").to_a + ["_"]
+    fake_key = Array.new(36).map { keyspace.sample }
+    # but mark it as allowed to test more of the program
+    (testpath/"test.txt").write("ghp_#{fake_key.join} # pragma: allowlist secret")
```

Comment:
```suggestion
    fake_key = keyspace.sample(36).join
    # but mark it as allowed to test more of the program
    (testpath/"test.txt").write("ghp_#{fake_key} # pragma: allowlist secret")
```

Reply:
That was my first pass but I decided that I wanted duplicates. [`Array#sample`](https://ruby-doc.org/core-2.6.8/Array.html#method-i-sample): 

> The elements are chosen by using random and unique indices into the array in order to ensure that an element doesn't repeat itself unless the array already contained duplicate elements.

But in re-reading the Array docs, I think this will help readability a little while maintaining the duplication in the randomness, FWIW.

```suggestion
    fake_key = Array.new(36) { keyspace.sample }.join
    # but mark it as allowed to test more of the program
    (testpath/"test.txt").write("ghp_#{fake_key} # pragma: allowlist secret")
```